### PR TITLE
Add support for node placement

### PR DESCRIPTION
**Pull request by vthapar**
Refer https://github.com/submariner-io/submariner-operator/pull/2530

### CODE REVIEW

File: pkg/hub/submarineragent/controller.go [hunk 1]
```diff
@@ -723,3 +752,54 @@ func (c *submarinerAgentController) updateBackupLabelOnBroker(ctx context.Contex
...
+	if len(nodePlacements) > 0 {
+		return nodePlacements, nil
+	}
+
+	/* No deployment config on managedclusteraddon, check default
+	   Ideally, we should get this from managedclusteraddon.status.configreferences but we don't for 2 reasons:
+	     1. There can be race condition between MCH controller adding addondeploymentconfig to managedclusteraddon.status
+	        vs when we read it, unless we watch for managedclusteraddon.status updates.
+	     2. We update managedclusteraddon.status. Not a good pattern to watch for updates on something we're
+	        updating as it will trigger extra cycles of update.
```
[the remark this answers is not in the document]
Comment (vthapar) [reply pkg/hub/submarineragent/controller.go:781]: Don't we need a separate watcher for status-only updates?

File: pkg/hub/submarineragent/controller.go [hunk 2]
```diff
@@ -723,3 +752,54 @@ func (c *submarinerAgentController) updateBackupLabelOnBroker(ctx context.Contex
...
 	return nil
 }
+
+func (c *submarinerAgentController) getAddonDeploymentConfigs(managedClusterAddon *addonv1alpha1.ManagedClusterAddOn) (
+	[]*addonv1alpha1.NodePlacement, error,
+) {
+	var nodePlacements []*addonv1alpha1.NodePlacement
+
+	for _, config := range managedClusterAddon.Spec.Configs {
+		if config.Resource == addonDeploymentConfigs {
```
[the remark this answers is not in the document]
Comment (vthapar) [reply pkg/hub/submarineragent/controller.go:762]: This should be fine at least for now as there is no other Group with same resource. We can revisit this later if it causes any bug in future.

[the remark this answers is not in the document]
Comment (tpantelis) [reply pkg/hub/submarineragent/controller.go:762]: yeah it's unlikely but doesn't hurt to check.

File: pkg/hub/submarineragent/controller.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (tpantelis) [reply pkg/hub/submarineragent/controller.go:781]: A watcher/informer gets all updates - there's no concept of a "status-only updates".

File: pkg/hub/manager.go [hunk 3]
```diff
@@ -201,6 +212,35 @@ func (o *AddOnOptions) RunControllerManager(ctx context.Context, controllerConte
 	return nil
 }
 
+func createClusterManagementAddon(ctx context.Context, client *addonclient.Clientset) error {
+	submarineerClusterMgmtAddon := &addonv1alpha1.ClusterManagementAddOn{
```
Comment (sridhargaddam) [inline pkg/hub/manager.go:216]: nit: submariner ;-)

**Review by sridhargaddam — APPROVED** [review]

**Pull request merged**